includes/class-wc-stripe-apple-pay-registration.php

```diff
@@ -241,34 +242,46 @@ public function process_apple_pay_verification() {
 	 * @since 4.0.6
 	 */
 	public function admin_notices() {
-		if ( ! $this->stripe_enabled ) {
+		if ( ! $this->stripe_enabled || ! $this->payment_request ) {
 			return;
 		}
 
 		if ( ! current_user_can( 'manage_woocommerce' ) ) {
 			return;
 		}
 
-		if ( $this->payment_request && ! empty( $this->apple_pay_verify_notice ) ) {
-			$allowed_html = array(
-				'a' => array(
-					'href'  => array(),
-					'title' => array(),
-				),
-			);
-
-			echo '<div class="error stripe-apple-pay-message"><p>' . wp_kses( make_clickable( $this->apple_pay_verify_notice ), $allowed_html ) . '</p></div>';
+		$empty_notice = empty( $this->apple_pay_verify_notice );
+		if ( $empty_notice && ( $this->apple_pay_domain_set || empty( $this->secret_key ) ) ) {
+			return;
 		}
 
 		/**
 		 * Apple pay is enabled by default and domain verification initializes
 		 * when setting screen is displayed. So if domain verification is not set,
 		 * something went wrong so lets notify user.
 		 */
-		if ( ! empty( $this->secret_key ) && $this->payment_request && ! $this->apple_pay_domain_set ) {
+		$allowed_html                      = array(
+			'a' => array(
+				'href'  => array(),
+				'title' => array(),
+			),
+		);
+		$error_from_stripe                 = $empty_notice ? '' : '<p>' . wp_kses( make_clickable( esc_html( $this->apple_pay_verify_notice ) ), $allowed_html ) . '.</p>';
+		$verification_failed_without_error = esc_html__( 'Apple Pay domain verification failed.', 'woocommerce-gateway-stripe' );
+		$verification_failed_with_error    = esc_html__( 'Apple Pay domain verification failed with the following error:', 'woocommerce-gateway-stripe' );
+		$check_log_text                    = sprintf(
 			/* translators: 1) HTML anchor open tag 2) HTML anchor closing tag */
-			echo '<div class="error stripe-apple-pay-message"><p>' . sprintf( __( 'Apple Pay domain verification failed. Please check the %1$slog%2$s to see the issue. (Logging must be enabled to see recorded logs)', 'woocommerce-gateway-stripe' ), '<a href="' . admin_url( 'admin.php?page=wc-status&tab=logs' ) . '">', '</a>' ) . '</p></div>';
-		}
+			esc_html__( 'Please check the %1$slogs%2$s for more details on this issue. Logging must be enabled to see recorded logs.', 'woocommerce-gateway-stripe' ),
+			'<a href="' . admin_url( 'admin.php?page=wc-status&tab=logs' ) . '">',
+			'</a>'
+		);
+
+		$notice_text = $empty_notice ? $verification_failed_without_error : $verification_failed_with_error;
+		echo '<div class="error stripe-apple-pay-message">';
```

Comment:
Not a big deal, just a thought: since it's all structure here, what do you think of having HTML here instead of PHP strings? And `$error_from_stripe` could be in a conditional maybe?

Reply:
That's a good idea I think 🤔 

I think I'll go ahead and do that 😄

Reply:
Done in 99bd5552585397ede97d5312ec814abcea834c63. What do you think?

Reply:
I like it 👍